feature-settings-impl/src/main/java/io/novafoundation/nova/feature_settings_impl/presentation/networkManagement/networkList/common/adapter/items/NetworkListNetworkRvItem.kt

```diff
@@ -0,0 +1,71 @@
+package io.novafoundation.nova.feature_settings_impl.presentation.networkManagement.networkList.common.adapter.items
+
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import coil.ImageLoader
+import io.novafoundation.nova.common.utils.images.Icon
+import io.novafoundation.nova.common.utils.images.setIcon
+import io.novafoundation.nova.common.utils.inflateChild
+import io.novafoundation.nova.common.utils.setDrawableStart
+import io.novafoundation.nova.common.utils.setTextColorRes
+import io.novafoundation.nova.common.utils.setTextOrHide
+import io.novafoundation.nova.feature_settings_impl.R
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.ChainId
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkImage
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkLabel
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkTitle
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkStatus
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkStatusShimmer
+import kotlinx.android.synthetic.main.item_network_settings.view.itemNetworkSubtitle
+
+class NetworkListNetworkRvItem(
+    val chainIcon: Icon,
+    val chainId: ChainId,
+    val title: String,
+    val subtitle: String?,
+    val chainLabel: String?,
+    val disabled: Boolean,
+    val status: ConnectionState?
+) : NetworkListRvItem {
+
+    override val id: String = chainId
+
+    class ConnectionState(
```

Comment:
Better to name it with some suffix, likle `ConnectionStateModel` otherwise the by-class search by `ConnectionState` will mix this with domain entity and make it harder to select right one just based on name